src/internet_identity/src/storage/tests.rs

```diff
@@ -94,11 +106,28 @@ fn should_serialize_first_record() {
 }
 
 #[test]
-fn should_serialize_subsequent_record_to_expected_memory_location() {
+fn should_serialize_first_record_v5() {
```

Comment:
I really like the approach of writing to storage with method you're testing and then reading directly from memory to make your assertions so the method under test is isolated, likewise for reading.

But I think it might also be useful to have a sanity check test where both storage read and write methods are used, so you can be sure that they are compatible with one another. Otherwise I think it's possible that the direct read/writes in the tests can deviate from the `Storage` implementation of reading and writing.

Reply:
On second thoughts, this is already covered in `should_keep_anchors_intact_when_migrating` in `src/internet_identity/tests/schema_migration_tests.rs`

Reply:
Thanks for pointing it out! This is a property that should not only be accidentally tested --> added tests for that.